packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/ClientModelPlugin.cs

```diff
@@ -36,15 +36,17 @@ public class ClientModelPlugin : CodeModelPlugin
         /// </summary>
         /// <param name="provider">The model type provider.</param>
         /// <param name="inputModel">The input model.</param>
-        public override IReadOnlyList<TypeProvider> GetSerializationTypeProviders(ModelProvider provider, InputModelType inputModel)
+        public override IReadOnlyList<TypeProvider> GetSerializationTypeProviders(TypeProvider provider, InputType inputType)
         {
-            if (inputModel.Usage.HasFlag(InputModelTypeUsage.Json))
+            switch (inputType)
             {
-                // Add MRW serialization type provider
-                return [new MrwSerializationTypeProvider(provider, inputModel)];
+                case InputModelType inputModel:
+                    return [new MrwSerializationTypeProvider(provider, inputModel)];
+                case InputEnumType inputEnum:
+                    return [EnumProvider.Create(inputEnum)];
```

Comment:
I don't think this should return an EnumProvider it should be returning the serialization that an enum needs.